db/post_migrate/20220704024901_migrate_settings_to_user_roles.rb

```diff
@@ -6,31 +6,26 @@ class MigrateSettingsToUserRoles < ActiveRecord::Migration[6.1]
   class UserRole < ApplicationRecord; end
 
   def up
-    owner_role     = UserRole.find_by(name: 'Owner')
-    admin_role     = UserRole.find_by(name: 'Admin')
-    moderator_role = UserRole.find_by(name: 'Moderator')
-    everyone_role  = UserRole.find_by(id: -99)
-
     min_invite_role  = Setting.min_invite_role
     show_staff_badge = Setting.show_staff_badge
 
-    if everyone_role
+    UserRole.find_by(id: -99).tap do |everyone_role|
       everyone_role.permissions &= ~::UserRole::FLAGS[:invite_users] unless min_invite_role == 'user'
       everyone_role.save
     end
```

Comment:
Wouldn't that cause `NoMethodError` if there is no such role? Same for the ones below.

Reply:
Hmm -- it doesn't raise that error ... but it DOES pass a `nil` into the block and then try to operate on it, which fails.

Updated this to use a different approach on this migration.